Mathlib/Algebra/MvPolynomial/SchwartzZippel.lean

```diff
@@ -0,0 +1,123 @@
+/-
+Copyright (c) 2023 Bolton Bailey. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bolton Bailey, Yaël Dillies
+-/
+import Mathlib.Algebra.MvPolynomial.Equiv
+import Mathlib.Algebra.Polynomial.Roots
+import Mathlib.Data.Fin.Tuple.Finset
+
+/-!
+# The Schwartz-Zippel lemma
+
+This file contains a proof of the
+[Schwartz-Zippel](https://en.wikipedia.org/wiki/Schwartz%E2%80%93Zippel_lemma) lemma.
+
+This lemma tells us that the probability that a nonzero multivariable polynomial over an integral
+domain evaluates to zero at a random point is bounded by the degree of the polynomial over the size
+of the field, or more generally, that a nonzero multivariable polynomial over any integral domain
+has a low probability of being zero when evaluated at points drawn at random from some finite subset
+of the field. This lemma is useful as a probabilistic polynomial identity test.
+
+## TODO
+
+* Generalize to subset of the ring being different for each variable.
+  * What does the theorem say in this case?
+  * Note that the Schwartz paper covers this case
+* Reexpress in terms of probabilities.
+* Then generalize to polynomials over arbitrary variable types
+* Write a tactic to apply this lemma to a given polynomial
+
+## References
+
+* [demillo_lipton_1978]
+* [schwartz_1980]
+* [zippel_1979]
+-/
+
+local prefix:100 "#" => Finset.card
+local notation:70 s:70 " ^^ " n:71 => Fintype.piFinset fun i : Fin n ↦ s i
+
+open Fin Finset Fintype MvPolynomial
+
+-- We want to be able to refer to `hp` and `hm`.
```

Comment:
I'm sorry, what do you mean with this?

Reply:
If I remove the `set_option` below, I must rename `hp` and `hm` to `_hp` and `_hm` in the statement